### ISSUE TITLE
[Enhancement] Movement Suggestions

### ISSUE THREAD

**Issue by Suis28** [alRex-U/ParCool#252]
Sliding is a cool movement technique that decreases the size of your hitbox, increases movement speed by a bit, and allows for 1 block movement. But it also locks the player's camera in the specified direction of the slide. And I have a few suggestions for the improvement of this movement technique.

- Slide Improvement/Suggestions:
_I think you should make slides:_
**1,** Go faster and/or longer the faster your movement speed is. Since slides often get outclassed by speed enhancing items, this idea is a good remedy to make the slide still relevant even when speed boosting items are present.
**2,** Have slide decay. Basically, the slide decelerates after a period of time, simulating the friction that is generated by sliding on the ground. Affected by suggestion 1.
**3,** Be configurable in the config files e.g. modifiers for sliding speed, slide decay, and slide cooldown.
**4,** Not lock the player's camera while still maintaining momentum in the specified direction of the slide.
**5,** Slow the player down after sliding to balance the speed the slide provides, and to simulate the "getting up" part of the slide

Although I understand that these suggestions may be hard to implement and code, but I know these tweaks will improve the sliding mechanic in this mod. Thank you for the consideration!

**Comment by alRex-U** [comment]
Thanks.

The 1st is already done before as bugfix.
Some people may feel the 2nd change annoying so I don't want to.

3rd suggestion may be not bad. Please let me think

And as for 4th suggestion, once it had behaved so in past but I disabled it. in the first place why I disable player rotation during sliding is that it was strange if not. It was quite strange to see Steve sliding in a curve at a constant speed.

About the last suggestion, I wonder if this is needed. I feel players gain enough slowing down from normal friction simulation. Further slowdowns may cause players to feel frustrated.

**Comment by Suis28** [comment]
Thank you for the consideration! I definitely look forward to the implementation of the 3rd suggestion. Also, you misinterpreted the 4th suggestion because of my poor phrasing, let me rephrase it. I think you should make slides not lock the player's camera while still maintaining momentum in the direction you started the slide. The slide will not change direction after you look away from the starting direction e.g. you're sliding north and you look west but the slide still continues to go north and isn't affected by the camera movement. Now that that is cleared up, I hope you take the new suggestion into consideration. Thank you and I'm a big fan of your work!

**Comment by alRex-U** [comment]
For now your 4th suggestion is done, it will applied in next update

**Comment by Suis28** [comment]
Thank you so much
